app/assets/javascripts/discourse/app/templates/components/badge-card.hbs

```diff
@@ -4,7 +4,22 @@
 {{#if badge.has_badge}}
   <a href={{url}} class="check-display status-checked">{{d-icon "check"}}</a>
 {{/if}}
-<div class="badge-contents">
+{{#if isFavorite}}
+  {{d-button
+    icon="star"
+    class="favorite"
+    action=(action "favorite")
```

Comment:
Im curious why don't we do this instead:

```suggestion
    action=(action onFavoriteClick)
```

and get rid of favorite action, same for next button